README.md

```diff
@@ -1913,6 +1913,143 @@ export type OrderList = {
 }
 ```
 
+### 11.7 TypeScript Utility Types
```

Comment:
Great ideas @lucioerlan 💡 

What do you think about [Typeof Type Operator](https://www.typescriptlang.org/docs/handbook/2/typeof-types.html) ?

I've been using those points who you showed. And I've been use too the typeof to the case typing utils, for instance

![Screenshot 2023-12-14 054838](https://github.com/juntossomosmais/frontend-guideline/assets/49894949/412271e2-509a-42c7-833b-002b6323b71c)

Typescript inferred the item as
```
{
    title: string;
    images: {
        name: string;
        legend: string;
    }[];
}
```

Reply:
tks @rogigs o/

it's beneficial because it ensures that your types are tight and evolve with your data structures.